### PR TITLE
Use Emscripten application

### PR DESCRIPTION
**Pull request by Squareys**
Hi @mosra !

As part of mosra/magnum#300, here are the ports of `player` and `ui-gallery`.

Best, Jonathan.

### CODE REVIEW

File: src/player/Player.cpp [hunk 1]
```diff
@@ -158,7 +162,7 @@ struct BaseUiPlane: Ui::Plane {
         stop{*this, {Ui::Snap::Right, play, ControlSize}, "Stop", Ui::Style::Danger},
         forward{*this, {Ui::Snap::Right, stop, HalfControlSize}, "»"},
         modelInfo{*this, {Ui::Snap::Top|Ui::Snap::Left, LabelSize}, "", Text::Alignment::LineLeft, 128, Ui::Style::Dim},
-        animationProgress{*this, {Ui::Snap::Right, forward, LabelSize}, "", Text::Alignment::LineLeft, 17}
+        animationProgress{*this, {Ui::Snap::Right, forward, LabelSize}, "", Text::Alignment::LineLeft, 18}
```
Comment (mosra) [inline src/player/Player.cpp]: No, this is my bug! :laughing:

File: src/player/CMakeLists.txt [hunk 2]
```diff
@@ -28,8 +28,14 @@ find_package(Magnum REQUIRED
     MeshTools
     Shaders
     SceneGraph
-    Trade
-    Sdl2Application)
+    Trade)
+
+
```
Comment (mosra) [inline src/player/CMakeLists.txt]: No more than one successive newline, please.

File: src/player/CMakeLists.txt [hunk 3]
```diff
@@ -71,7 +77,7 @@ endif()
 
 if(CORRADE_TARGET_EMSCRIPTEN)
     # globals yay :/
-    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -s 'EXTRA_EXPORTED_RUNTIME_METHODS=[\"ccall\"]' -s ALLOW_MEMORY_GROWTH=1")
+    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -s EXTRA_EXPORTED_RUNTIME_METHODS=[\"ccall\"] -s ALLOW_MEMORY_GROWTH=1")
```
Comment (mosra) [inline src/player/CMakeLists.txt:79]: Did this cause a problem on Windows or why is this?

**Review by mosra — COMMENTED** [review]
I guess I'll merge this after I fix the bug with animation progress being negative.

File: src/player/Player.cpp (hunk 1 above)
Comment (Squareys) [reply src/player/Player.cpp]: What do you mean? 😅

File: src/player/CMakeLists.txt (hunk 3 above)
Comment (Squareys) [reply src/player/CMakeLists.txt:79]: It caused a compile error "unclosed opened string list", yes. I suppose it has something to do with the `"` in the value...

**Comment by Squareys** [comment]
@mosra This is also ready for the next review!

File: src/player/Player.cpp (hunk 1 above)
Comment (Squareys) [reply src/player/Player.cpp]: > I guess I'll merge this after I fix the bug with animation progress being negative.

Ah

Comment (mosra) [reply src/player/Player.cpp]: Should be fixed with either e78edc7957ac6a4b972a36bd1594eb7abb6376de or ec63dd499ffbf05e89ee5a330f3e8faf58bbb28a that are currently in master. In particular, the animation time no longer goes negative for me, which triggered the assert here.

Can you rebase and confirm this is no longer needed? :)

**Review by mosra — COMMENTED** [review]
Looks good to me -- can you rebase on latest master and try w/o the assert hotfix?

File: src/player/Player.cpp (hunk 1 above)
Comment (Squareys) [reply src/player/Player.cpp]: Works! ✔️

**Comment by mosra** [comment]
Merged as a4caa32b801af8ff9610aa083ad3292e4575ea04, 4828c09464f801024ce304b9d0886e1b1581caea and eca456251ad36bee3241b01cf5855b77c34c7a4f, the apps are also shipped on https://magnum.graphics/showcase/ and working nicely. I omitted 3ab157466fa281e82b6a15cfb929bacd260d3337 because while that one fixes it for cmd.exe, it breaks it on the macOS shell (Travis complaining):

```
ERROR:root:a problem occurred in evaluating content after a "-s", specifically  EXTRA_EXPORTED_RUNTIME_METHODS=[ccall] . one possible cause of this is missing quotation marks (this depends on the shell you are running in; you may need quotation marks around the entire  EXTRA_EXPORTED_RUNTIME_METHODS=[ccall] , or on an individual element)
Traceback (most recent call last):
  File "/usr/local/Cellar/emscripten/1.37.36/libexec/em++", line 15, in <module>
    python_selector.run(emcc, profile=True)
  File "/usr/local/Cellar/emscripten/1.37.36/libexec/tools/python_selector.py", line 43, in run
    sys.exit(run_by_import(filename, main) if on_allowed_version() else run_by_subprocess(filename))
  File "/usr/local/Cellar/emscripten/1.37.36/libexec/tools/python_selector.py", line 13, in run_by_import
    return getattr(importlib.import_module(os.path.basename(filename)), main)()
  File "/usr/local/Cellar/emscripten/1.37.36/libexec/emcc.py", line 909, in run
    raise e
NameError: name 'ccall' is not defined
```

This worked correctly on my bash, but apparently not on macOS. Not sure what's the proper fix :sweat_smile:

**Pull request closed without merging**